crates/chain/src/tx_graph.rs

```diff
@@ -133,18 +134,18 @@ impl<A> Default for TxGraph<A> {
 
 /// A transaction node in the [`TxGraph`].
 #[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
-pub struct TxNode<'a, T, A> {
+pub struct TxNode<'a, T, AM> {
     /// Txid of the transaction.
     pub txid: Txid,
     /// A partial or full representation of the transaction.
     pub tx: T,
     /// The blocks that the transaction is "anchored" in.
-    pub anchors: &'a BTreeSet<A>,
+    pub anchors: &'a BTreeMap<Anchor, AM>,
```

Comment:
```suggestion
    pub anchors: Range<'a, Anchor, AM>,
```

I'm not sure if this will work? Getting a range from `TxGraph::anchors`.